api/src/lib.rs

```diff
@@ -153,16 +153,17 @@ impl PostieApi {
         }
 
         let res = req.send().await?;
-        let res_type = res.headers().get("content-type").unwrap().to_str().unwrap();
+        let res_headers = res.headers().clone();
+        let res_status = res.status().clone();
+        let res_type = res_headers.get("content-type").unwrap().to_str().unwrap();
+        let res_text = res.text().await?;
```

Comment:
So fun little bit I learned about reqwest - accessing the body either via text or json takes ownership of res since it requires ownership. To access the other bits off the response we need to clone the status code and headers before accessing the body.